Ticket update

xcoga opened the pull request:
test test

xcoga committed:
Ticket implementation

xcoga committed:
Update Ticket

xcoga committed:
update ticket 20/10/2022

aryans1204 commented:
FIrst PR looks good! Let me know once its complete, and I'll merge

xcoga committed:
Merge branch 'aryans1204:main' into main

xcoga committed:
added create_ticket_price method 

to create new ticket listings for recently added movies

aryans1204 commented on Ticket:
```diff
@@ -31,6 +33,20 @@ public class Ticket extends TicketAbs {
     	else return -1;
     }
+    
+    public void create_ticket_price(double[][][][]price_ar) {
```
Just a slight note on the naming convention, in Java we'll name methods createTicketPrice instead of using underscores. Functionally they're the same, but it just looks better on the Javadoc

aryans1204 commented on Ticket:
```diff
@@ -31,6 +33,20 @@ public class Ticket extends TicketAbs {
+    	cinema = sc.nextLine();
+    	System.out.println("Please enter showtime:");
+    	showtime = sc.nextInt();
+    	System.out.println("Please enter ticket price for child:");
```
Since in the document it's written that the ticket price is affected by factors like Age, Cinema type, Movie showtime etc, instead of taking the ticket price as input let's just set the initial ticket price automatically, based on the factors set above(we will call this method in the constructor when creating the ticket). For example, something along the lines of 

`if customer.age == child {
     ticket_price = movie.ticket_price[showtime] - movie.ticket_price[showtime]*0.2. //20% discount for child`

xcoga committed:
Merge branch 'aryans1204:main' into main

xcoga committed:
Created TicketController

writes to Ticket DB for new tickets created
writes to prices DB for default prices of movie. Ticket class will do the multipliers to calculate exact ticket price for each of the factors of movie type, client age, etc.

xcoga committed:
updated ticket class

added calculate price method, which will calculate exact prices for each ticket according to the factors (movie type, cinema type, age, etc.)

aryans1204 committed:
Update Ticket

aryans1204 commented on Ticket:
```diff
@@ -0,0 +1,97 @@
+    		multiplier = multiplier*1.5;
+    	}
+    	
+    	multiplier = multiplier * movieType * clientAge * cinemaType;
```
The overall ideology for calculating prices looks ok, but the movieType isnt an int, its an enum of `MovieType` defined in `MovieType.java`, so you'll have to create a switch statement to account for the different cases. Same thing with `CinemaType` defined in `CInemaAbs.java`.

aryans1204 commented on Ticket:
```diff
@@ -0,0 +1,97 @@
+
+    }
+    
+    public int[] getPricesFromDB() {
```
CRUD part looks good

aryans1204 commented on TicketController:
```diff
@@ -0,0 +1,127 @@
+	}
+	
+	public void updatePriceToDB(int moviename_index, int cinema_index,int showtime_index, int price) {
+		//do i have to use arraylist? 
```
No need, array suffices for `writeObject` since it takes as argument an `Object` which can be literally anything, even an array. However, the format of writing may differ. If you write an entire array, all the values will be included on a single line, whereas if you sue a for loop and write to each line, that will write the array vertically. Depends on what read method you're using.

aryans1204 reviewed (changes requested):
After the changes, it looks good enough to merge, at least as an initial build.

aryans1204 commented:
Additionally, also update the TicketAbs.java class with whatever additions you've made so others can refer in the end when you're done

xcoga committed:
Merge branch 'aryans1204:main' into main

xcoga committed:
updated multiplier

xcoga committed:
updated the abs

darrennjk commented on Ticket:
```diff
@@ -0,0 +1,140 @@
+		return prices;
+	}
+   
+    
```
Could you add a getter for customer (getCustomer) too? I think I will need it.

xcoga committed:
added getCustomer() method

xcoga committed:
added the method getCustomer()

aryans1204 commented on TicketAbs.java:
```diff
@@ -1,18 +1,29 @@
 
     public abstract Date getShowtime();  //method to return showtime of the ticket.
+    
+    public abstract Customer getCustomer(); //method to return customer of ticket
```
LGTM 🚀
